src/store/modules/aeternity.js

```diff
@@ -0,0 +1,207 @@
+/* globals Cypress */
+import {
+  MemoryAccount, Node, Universal, RpcAepp,
+} from '@aeternity/aepp-sdk/es';
+import BrowserWindowMessageConnection from '@aeternity/aepp-sdk/es/utils/aepp-wallet-communication/connection/browser-window-message';
+import Detector from '@aeternity/aepp-sdk/es/utils/aepp-wallet-communication/wallet-detector';
+import TIPPING_INTERFACE_V1 from 'tipping-contract/Tipping_v1_Interface.aes';
+import TIPPING_INTERFACE_V2 from 'tipping-contract/Tipping_v2_Interface.aes';
+import TIPPING_INTERFACE_V3 from 'tipping-contract/Tipping_v3_Interface.aes';
+import tippingContractUtil from 'tipping-contract/util/tippingContractUtil';
+import FUNGIBLE_TOKEN_CONTRACT from 'aeternity-fungible-token/FungibleTokenFullInterface.aes';
+import { BigNumber } from 'bignumber.js';
+import { IS_MOBILE_DEVICE } from '../../utils';
+
+export default {
+  state: {
+    sdk: null,
+    useSdkWallet: false,
+    useIframeWallet: false,
+    contractV1: null,
+    contractV2: null,
+    contractV3: null,
+    isTippingContractsInitialised: false,
+  },
+  mutations: {
+    setContractV1(state, contract) {
+      state.contractV1 = contract;
+    },
+    setContractV2(state, contract) {
+      state.contractV2 = contract;
+    },
+    setContractV3(state, contract) {
+      state.contractV3 = contract;
+    },
+    setSDK(state, { instance, rpcClient: getCurrentAccount }) {
+      state.sdk = instance;
+      if (getCurrentAccount) {
+        state.sdk.rpcClient = { getCurrentAccount };
+      }
+    },
+    useSdkWallet(state) {
+      state.useSdkWallet = true;
+    },
+  },
+  actions: {
+    async initTippingContractIfNeeded({
+      commit,
+      state: {
+        sdk,
+        isTippingContractsInitialised,
+      },
+    }) {
+      if (!sdk) throw new Error('Init sdk first');
+      if (isTippingContractsInitialised) return;
+      const [contractV1, contractV2, contractV3] = await Promise.all([
+        [TIPPING_INTERFACE_V1, process.env.VUE_APP_CONTRACT_V1_ADDRESS],
+        [TIPPING_INTERFACE_V2, process.env.VUE_APP_CONTRACT_V2_ADDRESS],
+        [TIPPING_INTERFACE_V3, process.env.VUE_APP_CONTRACT_V3_ADDRESS],
+      ].map(([tippingInterface, contractAddress]) => (contractAddress
+        ? sdk.getContractInstance(tippingInterface, { contractAddress })
+        : null)));
+      commit('isTippingContractsInitialised', true);
```

Comment:
I can't find where this mutation is defined

Reply:
thanks!